internal/bundlegen/generateapi.go

```diff
@@ -804,7 +803,7 @@ func parseKeyValuePairs(match string) map[string]interface{} {
 	match = strings.ReplaceAll(match, ")", "")
 
 	keyValuePairs := make(map[string]interface{})
-	keyValuePattern := `\b([A-Za-z0-9\-]+)\s*:\s*(\w+)\b`
+	keyValuePattern := `\b([A-Za-z0-9\-]+)\s*:\s*([A-Za-z0-9\-]+)\b`
```

Comment:
Now underscores in name don't work.

And still the step name for Quota isn't right when using underscore.
<UseQuotaConfigInAPIProduct stepName="Verify-API-Key-"/>

The VerifyAPIKey policy name is: Verify-API-Key-api_key